src/ejabberd_listener.erl

```diff
@@ -61,156 +53,50 @@ init(_) ->
 
 -spec start_listeners() -> 'ignore' | {'ok', {{_, _, _}, [any()]}}.
 start_listeners() ->
-    case mongoose_config:lookup_opt(listen) of
-        {error, not_found} ->
+    case mongoose_config:get_opt(listen) of
+        [] ->
             ignore;
-        {ok, Ls} ->
+        Ls ->
             Ls2 = lists:map(
-                fun({Port, Module, Opts}) ->
-                        case start_listener(Port, Module, Opts) of
+                fun(Listener) ->
+                        case start_listener(Listener) of
```

Comment:
Can simplify the case-of here, and do a lists:map nevertheless, it would loop over an empty list. That'd change the output of the function, but this function is only called in ejabberd_app and its return value is ignored, so it's all still fine 👌🏽

Reply:
I was thinking about it myself, I will do it.